### ISSUE TITLE
Supporting devices with SDL 2.0.0

### ISSUE THREAD

**Issue by peterkomar** [smartdevicelink/sdl_java_suite#1541]
### Bug Report
I'm using an old device with SDL 2.0 and I don't have the ability to upgrade to the latest. Looks like the last versions of android SDL doesn't support old SDL cores. 

##### Reproduction Steps
1. Set value for `MediaTrackTextField` it doesn't work with 4.12.x versions but work with 4.9.0 version
2. Getting permissions from the server can't process it with success with 4.12.x versions (it tries several times) but works with 4.9.0 version, permissions are downloaded but not applied correctly.  

##### OS & Version Information
* Android Version: 9
* SDL Android Version: 4.12.1
* Testing Against: [What you tested with to observe this behavior]

**Comment by bilal-alsharifi** [comment]
@peterkomar Thanks for reporting the issue. We fixed that bug in a [hotfix release (v4.12.2)]( https://github.com/smartdevicelink/sdl_java_suite/releases/tag/4.12.2) for the previous major version. We also included that fix in `develop` branch for the current major version (v5.0). We may also release a hotfix later (v5.0.1) for the current major version that includes the fix.

**Comment by peterkomar** [comment]
@bilal-alsharifi issue with MediaTrackTextField was fixed, thanks. 
But issues with permission still exist. In `RPCBuilder` app which was built with android sdk 4.9.0 to make it work I choose `Spec` as `Mobile_API_3.1.xml` instead of `Mobile_API_4.1.xml`. Is it something similar has 5.0 sdk or it detects automatically?

**Comment by bilal-alsharifi** [comment]
@peterkomar Thanks for testing the fixes. Can you elaborate more on the permission issue? do you mean the policy table updates don't work? Would it be possible to provide the exact reproductions steps for that issue, the expected behavior, and what is the wrong behavior that you are seeing?

**Comment by peterkomar** [comment]
![20201123_090200](https://user-images.githubusercontent.com/6721436/100873966-8a0c1900-34ac-11eb-824a-1229f7ae1595.jpg)

It's stuck in this state. I see some network traffic on my device from my app when retrieving permissions just starts, after that, I didn't see any network traffic until retrieving permissions didn't restart again.  On the older version of Android SDL 4.9.0 it was able to successfully finish. 

But the interesting thing I noticed on the older version (4.12.1 before the fix, also in 4.9.0) of Android SDL I was able to retrieve all vehicle data except tire pressure, it returned "Requested Vehicle Data is not available" but after the fix which you applied now I able to retrieve tire pressure as well. But I'm not totally sure if it related.

**Comment by joeygrover** [comment]
I retested with our Sync 2.0 TDK through the automatic process and the manual process. In both scenarios it appears to complete successfully. Please note that downloading the policy table and sending it to the module doesn't take very much time, however, the TDK takes around ~30 seconds to update the policy table on its end. At this point I don't believe there is much we can do on the library side to determine what is happening in your situation. I would recommend doing a factory reset of the module in case maybe the policy table became corrupt and it's unable to apply the update. See video of the manual process attached.

[Video capture of successful manual PTU.zip](https://github.com/smartdevicelink/sdl_java_suite/files/5638590/1541.zip)

**Comment by peterkomar** [comment]
@joeygrover Thanks for checking 👍

**Comment by peterkomar** [comment]
@joeygrover just tried again after a factory reset but the problem still the same. 
[12345.mov.zip](https://github.com/smartdevicelink/sdl_java_suite/files/5644639/12345.mov.zip) - it is a cut video  

I waiting for 2 mins. 

But when I run this app https://github.com/smartdevicelink/rpc_builder_app_android/releases lastest version I'm using this release https://github.com/smartdevicelink/rpc_builder_app_android/releases/download/1.4.0/RPCBuilder.apk with my app name and appid permissions were downloaded sucessfuly. 

My device is Sync 1.1 get 2 with software 5.11 and SDL 2.0.0


What version of Sync contains your Sync 2.0 TDK?

**Comment by peterkomar** [comment]
Just noticed that I'm using sdl sdk 5.0.0 will check on 4.12.2

**Comment by joeygrover** [comment]
So, just to double check, does that mean everything is working on your end now with 4.12.2?

**Comment by peterkomar** [comment]
I will give an answer later this week when I will have access to the vehicle.

**Comment by peterkomar** [comment]
Confirming that download permissions work with  4.12.2 on SDL 2.0 (I did a factory reset before the test ) but found the other issue:

 I the head unit I have only one checkbox `Body Info` and it is selected. Also, SYNC says that I have permissions to get:
  * fuel
  * odometer
  * vin
  * external Temperature
  * Tire Pressure

But when I make a request  I'm able to retrieve the odometer and vin only. Other data is not available (received response: vehicle data vas not available).

When I download permissions with sdl_java_suite 4.9.0 I'm able to retrieve all these data.

Could you check if data is available on your Sync 2.0 TDK after download permissions? 

Also found when I check permissions with 
```
sdlManager.getPermissionManager().isPermissionParameterAllowed(FunctionID.GET_VEHICLE_DATA, GetVehicleData.<param>);
```

it always returns false for all of them but actually, I'm able to get some of the data. when I call for example:
```
Integer odometer = ((GetVehicleDataResponse) response).getOdometer();
``` 
I'm able to get the odometer but android `PermissionManager` says that this data is not available.

**Comment by joeygrover** [comment]
I have confirmed that the policy table update occurs correctly with our unit. I was able to grant user consent when activating an app that requested vehicle data. After this consent, the app received an OnPermissionChange that contained the correct state of permissions. The permission manager then returns true for the following:

```java
PermissionManager permissionManager = sdlManager.getPermissionManager();
Boolean bodyInfoAllowed = permissionManager.isPermissionParameterAllowed(FunctionID.GET_VEHICLE_DATA, GetVehicleData.KEY_BODY_INFORMATION);
Log.d(TAG, "Body info allowed: " + bodyInfoAllowed);
```

The app is then able to retrieve the body information successfully. My only suggestion would be to make sure you are using the correct app name and app ID that was provided to Ford for access.

At this point I'm really not sure how much more we are going to be able to help find the issue. This might be a better time to talk directly with your Ford contact. Of course we will assist if we can, but I don't see where the library would be at fault at this point.
